Source/Evolve/WarpXEvolve.cpp

```diff
@@ -145,6 +147,7 @@ WarpX::Evolve (int numsteps)
         if (cur_time + dt[0] >= stop_time - 1.e-3*dt[0] || step == numsteps_max-1) {
             // At the end of last step, push p by 0.5*dt to synchronize
             UpdateAuxilaryData();
+            FillBoundaryAux(guard_cells.ng_UpdateAux);
```

Comment:
Note: adding this call to `FillBoundaryAux` causes the benchmark of the CI test `momentum-conserving-gather` to slightly change (see commit https://github.com/ECP-WarpX/WarpX/pull/1468/commits/1a51405748bedea66c88e57ea1e779f387728d89).